src/back/spv/writer.rs

```diff
@@ -809,6 +840,43 @@ impl Writer {
         id
     }
 
+    fn get_type_inner<'a>(
+        &self,
+        ty_arena: &'a crate::Arena<crate::Type>,
+        lookup_ty: LookupType,
+    ) -> MaybeOwned<'a, crate::TypeInner> {
+        match lookup_ty {
+            LookupType::Handle(handle) => MaybeOwned::Borrowed(&ty_arena[handle].inner),
+            LookupType::Local(local_ty) => match local_ty {
+                LocalType::Scalar { kind, width } => {
+                    MaybeOwned::Owned(crate::TypeInner::Scalar { kind, width })
+                }
+                LocalType::Vector { size, kind, width } => {
+                    MaybeOwned::Owned(crate::TypeInner::Vector { size, kind, width })
+                }
+                LocalType::Pointer { base, class } => {
+                    let storage_class = match class {
+                        spirv::StorageClass::UniformConstant => crate::StorageClass::Handle,
```

Comment:
It seems backwards that we map from SPIR-V into IR here. Can we make it so `LocalType::Pointer` stores IR's StorageClass instead? That would make the conversion only one-way, and we could have a single helper conversion function.

Reply:
Yes agreed! We already have a mapper from `crate::StorageClass` to `spirv::StorageClass`, so that will be easier, and more maintainable. 

Only the instructions would require the `spirv::StorageClass`, rest can be the `crate::StorageClass`.